api/src/api/notes.js

```diff
@@ -80,23 +80,46 @@ router.post(
   }),
 );
 
+const getVersionHistory = async (id) => {
+  const note = await Note.findById(id);
+  return note.metaData.versionHistory;
+};
+
 router.put(
   '/:notesId',
   requireRegistered,
   validateEditability,
   validateReqParams,
   errorWrap(async (req, res) => {
     let data = { ...req.body };
-    const updatedNote = await Note.findByIdAndUpdate(
-      req.params.notesId,
-      { $set: req.body },
-      { new: true },
-    );
-    res.status(200).json({
-      success: true,
-      message: 'Note successfully updated',
-      data: updatedNote,
-    });
+    // Get current version history and append latest edit
+    const currentVersionHistory = await getVersionHistory(req.params.notesId);
+    if (currentVersionHistory) {
+      currentVersionHistory.push({
+        date: Date.now(),
+        action: Note.actions.EDITED,
+        memberID: req.user._id,
+      });
+
+      data.metaData.versionHistory = currentVersionHistory;
```

Comment:
Should you return here or raise an error since version history would be null or empty?

Reply:
He's setting version history though, right? So that shouldnt be a concern